### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-input

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2054]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-input component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-input can be found here: [sd-input Notes](https://www.notion.so/matuzo/sd-input-1baf3f6e1bc08100b262d88e6a242c24).

------------------------------------------

**Problem:** Invalid fields don't semantically indicate that they're invalid.
**Solution:** Put `aria-invalid="true"` on invalid fields.

**Problem:** The “Show password”-button is not accessible via the keyboard.
**Solution:** Remove the `tabindex`-attribute.

## Notes
- Instead of (or in addition to) changing the label, you could also toggle the `aria-pressed` attribute to indicate whether the password is visible.
- Please note that `maxlength` doesn't indicate the maximum number of characters to screen readers. If there's no help text, it can be confusing.
- Consider noting in the docs that the placeholder is not a replacement for the label but only an addition.
- Disabled inputs: See [[sd-button](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21)](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21) . Everything I wrote there about disabling form elements applies here as well.

## Accessibility information
- *Ensure that focus moves logically between input fields and other form elements. When an input field is focused, it should be clearly indicated.*
**”logically” can be misleading. Consider using “in sequential order” instead.**
- *If an input field contains an error, ensure that it is announced by screen readers. Use aria-invalid="true" and aria-describedby to link the input field to the error message.*
**Isn’t that something the component already does for me?**
- Use the right type for the given purpose (email, url, etc.)
- Use the `autocomplete` attribute to help browsers auto-fill fields
- In most cases use visually visible labels.

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by smfonseca** [comment]
@coraliefeil the tooltip for `visually-disabled` should also show up on hover like in the button, right? It should be the default behaviour on all components with this attribute.

**Comment by smfonseca** [comment]
@coraliefeil @MartaPintoTeixeira can you please check the notes and a11y information? 

First point of notes can be disregarded since I've already addressed it in code. 
Second point on a11y info should be removed as suggested.
